src/main/java/com/treasure/hunt/game/GameEngine.java

```diff
@@ -232,11 +221,35 @@ protected void verifyHint(Hint hint, Point treasurePosition) {
         }
     }
 
+    /**
+     * Verifies whether the performed {@link Movement} {@code movement} by the {@link Searcher} followed the rules.
+     *
+     * @param movement                which gets verified
+     * @param initialSearcherPosition initial searcher position
+     * @throws IllegalArgumentException when the {@link Movement} is not valid.
+     */
+    protected void verifyMovement(Movement movement, Point initialSearcherPosition) {
+        if (movement.getStartingPoint().getX() != initialSearcherPosition.getX() ||
+                movement.getStartingPoint().getY() != initialSearcherPosition.getY()) {
+            throw new IllegalArgumentException("Searcher stands last at " + initialSearcherPosition +
+                    " but continues his movement from " + movement.getStartingPoint());
+        }
+        for (GeometryItem geometryItem : movement.getPoints()) {
+            if (outOfMap(geometryItem.getGeometry().getCoordinate())) {
+                throw new IllegalArgumentException("Searcher left the playing area: " +
```

Comment:
U never tell the searcher and hider what the WIDTH and HEIGHT are but u throw Exceptions if they go outside of it seems weired to me

Reply:
Maybe we can look into it and see if we come up with a simpilier solution 🤔

Reply:
This is right, i'm going to fix this in PR #107